common/hyperscaler/rules/evaluator.go

```diff
@@ -0,0 +1,46 @@
+package rules
+
+import (
+	"fmt"
+
+	"github.com/kyma-project/kyma-environment-broker/internal/broker"
+)
+
+type Evaluator struct {
+	rules map[string][]*Rule
+}
+
+func NewEvaluator(rules []*Rule) *Evaluator {
+	evaluator := &Evaluator{
+		rules: make(map[string][]*Rule),
+	}
+
+	for _, rule := range rules {
+		if _, exists := evaluator.rules[rule.Plan]; !exists {
+			evaluator.rules[rule.Plan] = make([]*Rule, 0)
+		}
+		evaluator.rules[rule.Plan] = append(evaluator.rules[rule.Plan], rule)
+	}
+
+	return evaluator
+}
+
+/**
+ * Evaluate rules and output search labels.
+ */
+func (e *Evaluator) Evaluate(matchableAttributes *MatchableAttributes) ([]*Rule, error) {
+
+	if _, ok := broker.PlanIDsMapping[matchableAttributes.Plan]; !ok {
```

Comment:
Again - IMHO we should not base only on this map.

Reply:
Removed Evaluator: https://github.com/ralikio/kyma-environment-broker/commit/237cd1ce131d52d52afd2631a797cb75096903e3